src/documentation/io/authoring-overview.md

```diff
@@ -10,35 +10,80 @@ permalink: /documentation/io/authoring-overview/
+3. As you're working on your IO transform, see the [PTransform Style Guide]({{ site.baseurl }}/contribute/ptransform-style-guide/) for specific information about writing I/O Transforms.
+
 
 ## Read transforms
-TODO
+Read transforms take data from outside of the Beam pipeline and produce PCollections of data.
+
+For data stores or files types where the data can be read in parallel, you can think of the process as a mini-pipeline. This often consists of two steps:
+1. Splitting the data into parts to be read in parallel
+2. Reading from each of those parts
+
+Each of those steps will be a `ParDo`, with a `GroupByKey` in between. The `GroupByKey` is an implementation detail, but for most runners it allows the runner to use different numbers of workers for:
+* Determining how to split up the data to be read into chunks - this will likely occur on very few workers
+* Reading - will likely benefit from more workers
+
+The `GroupByKey` will also allow Dynamic Work Rebalancing to occur (on supported runners).
+
+Here are some examples of read transform implementations that use the "reading as a mini-pipeline" model when data can be read in parallel:
+* **Reading from a file glob** - For example reading all files in "~/data/**"
+  * Get File Paths `ParDo`: As input, take in a file glob. Emit a PCollection of strings, each of which is a file path.
+  * Reading `ParDo`: Given the PCollection of file paths, read each one, emitting a PCollection of records.
+* **Reading from a NoSQL Database** (eg Apache HBase) - these databases often allow reading from ranges in parallel.
+  * Determine Key Ranges `ParDo`: As input, receive connection information for the database and the key range to read from. Output a PCollection of key ranges that can be read in parallel efficiently.
+  * Read Key Range `ParDo`: Given the PCollection of key ranges, read the key range, emitting a PCollection of records.
+
+For data stores or files where reading cannot occur in parallel, reading is a simple task that can be accomplished with a single `ParDo`+`GroupByKey`. For example:
+* **Reading from a database query** - traditional SQL database queries often can only be read in sequence. The `ParDo` in this case would establish a connection to the database and read batches of records, emitting a PCollection of those records.
+* **Reading from a gzip file** - a gzip file has to be read in order, so it cannot be parallelized. The `ParDo` in this case would open the file and read in sequence, emitting a PCollection of records from the file.
+
+
+### When to implement using the `Source` API
+The above discussion is in terms of `ParDo`s - this is because `Source`s have proven to be tricky to implement. At this point in time, **if you're not reading from a file, it is not recommended that you implement a `Source` if you can get away with a `ParDo`**. A class derived from `FileBasedSource` is often the best option when reading from files.
```

Comment:
Rephrase with fewer negation: use Source only if you are reading from files and can not get away with simply using a ParDo.